source/agora/node/BlockStorage.d

```diff
@@ -918,17 +918,23 @@ public class MemBlockStorage : IBlockStorage
     private IndexHash hash_idx;
 
     /// Ctor
-    public this ()
+    public this (const(Block)[] blocks = null)
     {
         this.height_idx = new IndexHeight();
         this.hash_idx = new IndexHash();
+
+        if (blocks.length == 0)
+            this.saveBlock(GenesisBlock);
+        else
+        {
+            foreach (const ref block; blocks)
+                this.saveBlock(block);
+        }
     }
 
     /// No-op: MemBlockStorage does no I/O
     public override bool load ()
     {
-        if (this.blocks.length == 0)
-            this.saveBlock(GenesisBlock);
```

Comment:
I would add a boolean in the new class, or at least something that checks we don't accidentally forget to initialize the BlockStorage from our code. [It happened before](https://github.com/bpfkorea/agora/commit/c7b90eb9483607d900fef63c3a7b63b2cae2b5dc)

Reply:
Ah then I'll move the loading into this method.